test/unit/account/gateway_test.rb

```diff
@@ -126,4 +126,11 @@ class Account::GatewayTest  < ActiveSupport::TestCase
     buyer.payment_detail.payment_method_id = 'pm_1I5s3n2eZvKYlo2CiO193T69'
     assert_instance_of ActiveMerchant::Billing::StripePaymentIntentsGateway, buyer.provider_payment_gateway
   end
+
+  test 'unacceptable_payment_gateway?' do
+    assert FactoryBot.create(:simple_account, payment_gateway_type: :bogus).unacceptable_payment_gateway?
+    assert FactoryBot.create(:simple_account, payment_gateway_type: nil).unacceptable_payment_gateway?
+
+    assert_not FactoryBot.create(:simple_account, payment_gateway_type: :braintree_blue).unacceptable_payment_gateway?
+  end
```

Comment:
it is a nice refactoring. When I asked in the past about such test though, I thought about where would we actually see something like this. More of an integration test. It is ok to keep as is though if you have more useful other things to do.

Reply:
Yeah, this is not supposed to be the test coverage I intended to implement! 😅